shell/platform/darwin/macos/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -0,0 +1,98 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+
+#include "flutter/shell/platform/darwin/macos/framework/Headers/FlutterEngine.h"
+#include "flutter/shell/platform/darwin/macos/framework/Source/FlutterDartProject_Internal.h"
+#include "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h"
+#include "flutter/testing/testing.h"
+#import "third_party/ocmock/Source/OCMock/OCMock.h"
+
+namespace flutter::testing {
+
+// Returns a mock FlutterViewController that is able to work in environments
+// without a real pasteboard.
+id mockViewController() {
+  NSString* fixtures = @(testing::GetFixturesPath());
+  FlutterDartProject* project = [[FlutterDartProject alloc]
+      initWithAssetsPath:fixtures
+             ICUDataPath:[fixtures stringByAppendingString:@"/icudtl.dat"]];
+  FlutterViewController* viewController = [[FlutterViewController alloc] initWithProject:project];
+
+  // Mock _pasteboard so that this test will work in environments without a
+  // real pasteboard.
+  id pasteboardMock = OCMClassMock([NSPasteboard class]);
+  __block NSString* clipboardString = @"";
+  OCMStub([pasteboardMock setString:[OCMArg any] forType:[OCMArg any]])
+      .andDo(^(NSInvocation* invocation) {
+        [invocation getArgument:&clipboardString atIndex:2];
+      });
+  OCMExpect([pasteboardMock stringForType:[OCMArg any]]).andDo(^(NSInvocation* invocation) {
+    [invocation setReturnValue:&clipboardString];
+  });
+  id viewControllerMock = OCMPartialMock(viewController);
+  OCMStub([viewControllerMock _pasteboard]).andReturn(pasteboardMock);
+  return viewControllerMock;
+}
+
+TEST(FlutterViewControllerTest, HasStringsWhenPasteboardEmpty) {
+  id viewControllerMock = mockViewController();
+
+  // Call setData to make sure that the pasteboard is empty.
```

Comment:
Your test controls the fake clipboard; there is no reason to involve `Clipboard.setData` in the test of Clipboard.hasStrings.

It would be good to have a test of `setData`, but that should be a separate test.

Reply:
Done, now I just set up the mock with some given pasteboard string instead of letting the test call setString.